uhubctl.c

```diff
@@ -790,28 +790,30 @@ int main(int argc, char *argv[])
         );
         exit(1);
     }
-    int i;
-    for (i=0; i<hub_count; i++) {
-        if (hubs[i].actionable == 0)
-            continue;
-        printf("Current status for hub %s [%s]\n",
-            hubs[i].location, hubs[i].description
-        );
-        print_port_status(&hubs[i], opt_ports);
-        if (opt_action == POWER_KEEP) { /* no action, show status */
-            continue;
-        }
-        struct libusb_device_handle * devh = NULL;
-        rc = libusb_open(hubs[i].dev, &devh);
-        if (rc == 0) {
-            /* will operate on these ports */
-            int ports = ((1 << hubs[i].nports) - 1) & opt_ports;
-            int k; /* k=0 for power OFF, k=1 for power ON */
-            for (k=0; k<2; k++) { /* up to 2 power actions - off/on */
-                if (k == 0 && opt_action == POWER_ON )
-                    continue;
-                if (k == 1 && opt_action == POWER_OFF)
-                    continue;
+    int k; /* k=0 for power OFF, k=1 for power ON */
+    for (k=0; k<2; k++) { /* up to 2 power actions - off/on */
+        if (k == 0 && opt_action == POWER_ON )
+           continue;
+        if (k == 1 && opt_action == POWER_OFF)
+           continue;
+        if (k == 1 && opt_action == POWER_CYCLE)
+           sleep_ms(opt_delay * 1000);
```

Comment:
This sleep should remain around line 849, like this:

    if (k == 0 && opt_action == POWER_CYCLE)
        sleep_ms(opt_delay * 1000);

Its debatable at what stage delay should be introduced - after power off (k==0) or just before power on (k==1).
But, I want to eventually fix some hubs slow power off (e.g. current extra wait 150ms for USB3 hubs). Proper solution would be to turn off and wait until power was actually turned off. This will need this delay sleep to happen right after, but still when k==0.